speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionMakerTests.cpp

```diff
@@ -55,3 +60,91 @@ TEST_CASE(DecisionMakerTests, ChangeSettingsTestNoIntervention, ChangeSettingsTe
+    chdir(SD_DATADIR_SRC);
+    ASSERT_TRUE(FindFileDirectory(findfilepath, "blackbox.exe"));
+    std::string bbPath = findfilepath.append("\\blackbox.exe");
+    p_decisionMaker.Initialize(&car, &situation, &track, bbPath, true);
+
+    SocketBlackBoxMock* mockCheck = dynamic_cast<SocketBlackBoxMock*>(&p_decisionMaker.BlackBox);
+    BlackBoxData* blackboxDataMock = mockCheck->GetBlackBoxData();
+    FileDataStorageMock* storage = p_decisionMaker.GetFileDataStorage();
+
+    ASSERT_TRUE(storage->EnvironmentVersion == track.version);
+    ASSERT_TRUE(blackboxDataMock->Car.pub.speed == car.pub.speed);
+    ASSERT_TRUE(blackboxDataMock->Situation.deltaTime == situation.deltaTime);
+}
+
+/// @brief Runs the initialize test function
+TEST(DecisionMakerTests, InitializeTest)
+{
+    TDecisionMaker decisionMaker;
+    InitializeTest(decisionMaker);
+}
+
+/// @brief				 Tests if the data collection settings can be set correctly
+/// @param  p_dataToStore data settings that will be set.
+void SetDataCollectionSettingsTest(DataToStore p_dataToStore)
+{
+    TDecisionMaker decisionMaker;
+    decisionMaker.SetDataCollectionSettings(p_dataToStore);
+    ASSERT_TRUE(decisionMaker.Config.GetDataCollectionSetting().CarData == p_dataToStore.CarData);
+    ASSERT_TRUE(decisionMaker.Config.GetDataCollectionSetting().EnvironmentData == p_dataToStore.EnvironmentData);
+    ASSERT_TRUE(decisionMaker.Config.GetDataCollectionSetting().HumanData == p_dataToStore.HumanData);
+    ASSERT_TRUE(decisionMaker.Config.GetDataCollectionSetting().InterventionData == p_dataToStore.InterventionData);
+    ASSERT_TRUE(decisionMaker.Config.GetDataCollectionSetting().MetaData == p_dataToStore.MetaData);
+}
+
+/// @brief				 Performs the data collection test with the given parameters
+/// @param  p_environmentData, p_carData, p_humanData , p_interventionData, p_metaData are the individual data settings
+void DoSetDataCollectionTest(bool p_environmentData, bool p_carData, bool p_humanData, bool p_interventionData, bool p_metaData)
+{
+    DataToStore dataSettings = {
+        p_environmentData,
+        p_carData,
+        p_humanData,
+        p_interventionData,
+        p_metaData};
+    SetDataCollectionSettingsTest(dataSettings);
+}
+
+/// @brief Does the SetDataCollectionSettingsTest with all possible boolean combinations
+BEGIN_TEST_COMBINATORIAL(DecisionMakerTests, SetDataCollectionSettingsTestAll)
+bool arr[2]{false, true};
+END_TEST_COMBINATORIAL5(DoSetDataCollectionTest, arr, 2, arr, 2, arr, 2, arr, 2, arr, 2);
+
+/// @brief Tests if the RaceStop function is correctly implemented and if it uses the correct path
+TEST(DecisionMakerTests, RaceStopTest)
+{
+    TDecisionMaker decisionMaker;
+    InitializeTest(decisionMaker);
+    chdir(SD_DATADIR_SRC);
+    ASSERT_NO_THROW(decisionMaker.RaceStop());
+    std::experimental::filesystem::path path = *static_cast<std::experimental::filesystem::path*>(VariableStore::GetInstance().Variables[0]);
```

Comment:
why cast it to a pointer and then get the value from the pointer?
```suggestion
    std::experimental::filesystem::path path = static_cast<std::experimental::filesystem::path>(VariableStore::GetInstance().Variables[0]);
```

Reply:
No because it is a void pointer